### ISSUE TITLE
Unable to use tag_names in site24x7_website_monitor

### ISSUE THREAD

**Issue by Th0masL** [site24x7/terraform-provider-site24x7#97]
Hi,

We are trying to create a monitor and apply tags by specifying `tag_names`, but it's not working.

Only the `tag_ids` approach is working.

My tags are using the following values :
```
# result from the API call https://www.site24x7.com/api/tags
    {
      "tag_value": "nginx-prod",
      "tag_name": "service",
      "tag_type": 1,
      "tag_color": "#E55445",
      "tag_id": "29977100xxxxxxxxxx"
    },
    {
      "tag_value": "nginx-dev",
      "tag_name": "service",
      "tag_type": 1,
      "tag_color": "#B7DA9E",
      "tag_id": "29977100xxxxxxxxxx"
    },
    {
      "tag_value": "nginx-stage",
      "tag_name": "service",
      "tag_type": 1,
      "tag_color": "#FCEA8B",
      "tag_id": "29977100xxxxxxxxxx"
    },
```

So I'm trying to use the following syntax in Terraform module, but it's not working
```
  tag_names = [
    "service:nginx-prod"
  ]
```

What is the expected syntax ?

Thanks

Thomas

**Comment by velimir** [comment]
It doesn't look like the tag_names are even used in website monitoring module: https://github.com/site24x7/terraform-provider-site24x7/blob/b1d27a459bdbb282f16c55e6274da6c5601b5946/site24x7/monitors/website.go#L542-L545, only `tag_ids`.

**Comment by velimir** [comment]
Unfortunately one can't use tag ids either as id is not exposed for the tag resource. Here's an attempt to address this: https://github.com/site24x7/terraform-provider-site24x7/pull/98

**Comment by jim-billy** [comment]
Hi Thomas,

Please try giving the value present in the "tag_name"(in the API response) eg:

tag_names = [
    "service"
  ]

Please bear with us till we support data sources for fetching tag_ids.

Thanks & Regards,
Jim

**Comment by jim-billy** [comment]
> It doesn't look like the tag_names are even used in website monitoring module:
> 
> https://github.com/site24x7/terraform-provider-site24x7/blob/b1d27a459bdbb282f16c55e6274da6c5601b5946/site24x7/monitors/website.go#L542-L545
> , only `tag_ids`.

Please refer https://github.com/site24x7/terraform-provider-site24x7/blob/b1d27a459bdbb282f16c55e6274da6c5601b5946/site24x7/monitors/website.go#L619-L624

**Comment by velimir** [comment]
@jim-billy my apologies, haven't noticed where the tags are applied.

Unfortunately the mentioned block doesn't take into account tag values. Having the following tags defined `service:foo`, `service:bar` and `service:quz` and using the suggested TF block when for the monitor definition:

```HCL
tag_names = [
"service"
]
```

Would result in a monitor that would be tagged with 3 tags listed above.

@jim-billy Could you please confirm that it's the case? And if so, is it expected behavour? What I'm trying to understand what is the primary use-case of the tags in the site24x7 definition of it.

**Comment by jim-billy-zoho** [comment]
@velimir Our apologies for not considering tag values. Will fix the issue and update here.

**Comment by jim-billy-zoho** [comment]
@Th0masL @velimir 
To fetch the tags, please use the `site24x7_tag` data source available in our latest version 1.0.24 - https://registry.terraform.io/providers/site24x7/site24x7/latest/docs/data-sources/tag
